coverage/docs/design.md

```diff
@@ -0,0 +1,139 @@
+comment access to the repo it need to be run on. If the repo is private, it also need read access. 
+  
+After the robot account is created, download the github token and supply the path to the token 
+file to code coverage binary, as the value for parameter "github-token"
+
+#Usage with prow
+
+Prow can be used as the system to handle Github events mentioned in the two workflows. We can add a pre-submit prow job that is triggered by any new commit to a PR to run test coverage on the new build to compare it with the master branch and previous commit for pre-submit coverage. We can add a post-submit prow job that is triggered by merge events, to run test coverage when ever there is a merge on the target branch. 
+
+In addition, at the end of each workflow, prow copies the artifacts directory to gcs bucket
+
+
+##Prow Configuration File
+As mentioned earlier, we use configuration file to store repository specific information. Below is an example that contains the args that will be supplied to the coverage container
+```
+  - name: pull-knative-serving-go-coverage
+    labels:
+      preset-service-account: "true"
+    always_run: true
+    optional: true
+    decorate: true
+    clone_uri: "git@github.com:knative/serving.git"
+    ssh_key_secrets:
+    - ssh-knative
+    spec:
+      containers:
+      - image: gcr.io/knative-tests/test-infra/coverage:latest
+        imagePullPolicy: Always
+        command:
+        - "/coverage"
+        args:
+        - "--postsubmit-gcs-bucket=knative-prow"
+        - "--postsubmit-job-name=post-knative-serving-go-coverage"
+        - "--artifacts=$(ARTIFACTS)" # folder to store artifacts, such as the coverage profile
+        - "--profile-name=coverage_profile.txt"
+        - "--cov-target=./pkg/" # target directory of test coverage
+        - "--cov-threshold-percentage=50" # minimum level of acceptable presubmit coverage on a per-file level
+        - "--github-token=/etc/github-token/token"
+        volumeMounts:
+        - name: github-token
+          mountPath: /etc/github-token
+          readOnly: true
+      volumes:
+      - name: github-token
+        secret:
+          secretName: covbot-token
+```
+
+#Acceptance Criteria
+##Presubmit Workflow
+- When user made a new commit, if any file in the commit has a coverage change, post a covbot comment concluding the coverage change
+- When user made a new commit, if non of the files in the commit has a coverage change, do not post any comment
+- The covbot comment should have the following information on each file with a coverage change:
+  - file name
+  - old coverage (coverage before any change in the PR)
+  - new coverage (coverage after applied all changes in the PR)
+  - change the coverage
+- After each new posting, any previous posting by covbot should be removed
+- When coverage threshold is enforced, block PR from merging
+
+##Periodical Workflow
```

Comment:
Does only periodic triggering create jUnit XML?

Reply:
Periodical Job is removed. Both pre- and post-submit create the XML